File: cms/src/Astrid/Tech/InputSchema/Tag.hs
module Astrid.Tech.InputSchema.Tag () where

import Data.Aeson
import GHC.Generics

data TagOverride = TagOverride
  { slug :: String,
    name :: String
  }
  deriving (Generic)

instance FromJSON TagOverride

data ColorGroup = ColorGroup
  { backgroundColor :: String,
    color :: Maybe String,
    tags :: TagOverride
  }
  deriving (Generic)

instance FromJSON ColorGroup

type TagOverrideSheet = [ColorGroup]

File: astrid_tech_frontend/lib/entity/Tag.ts
import { Entity, PrimaryGeneratedColumn, Column, JoinTable } from "typeorm";

@Entity()
export class Tag {
  @PrimaryGeneratedColumn()
  id!: number;

  @Column({ nullable: false })
  slug!: string;

  @Column({ nullable: false })
  name!: string;

  @Column({ nullable: false })
  backgroundColor!: string;

  @Column({ nullable: false })
  color!: string;
}
